.github/workflows/ci_mlir_plugin.yml

```diff
@@ -0,0 +1,218 @@
+  #          echo "CXX=g++-14" >> $GITHUB_ENV
+  #          echo "MLIR_DIR=/usr/lib/llvm-${{ matrix.llvm-version }}/lib/cmake/mlir" >> $GITHUB_ENV
+  #          echo "LLVM_DIR=/usr/lib/llvm-${{ matrix.llvm-version }}/lib/cmake/llvm" >> $GITHUB_ENV
+  #
+  #      # Install llvm and ninja
+  #      - name: Install llvm@${{ matrix.llvm-version }} and Ninja via Homebrew
+  #        if: runner.os == 'macOS'
+  #        run: brew install llvm@${{ matrix.llvm-version }} ninja
+  #
+  #      # Set compiler and environment variables
+  #      - name: Set compiler and CMake environment variables
+  #        if: runner.os == 'macOS'
+  #        run: |
+  #          if [ "$(uname -m)" = "x86_64" ]; then
+  #            LLVM_PREFIX="/usr/local/opt/llvm@${{ matrix.llvm-version }}"
+  #          else
+  #            LLVM_PREFIX="/opt/homebrew/opt/llvm@${{ matrix.llvm-version }}"
+  #          fi
+  #
+  #          echo "CC=$LLVM_PREFIX/bin/clang" >> $GITHUB_ENV
+  #          echo "CXX=$LLVM_PREFIX/bin/clang++" >> $GITHUB_ENV
+  #          echo "LLVM_DIR=$LLVM_PREFIX/lib/cmake/llvm" >> $GITHUB_ENV
+  #          echo "MLIR_DIR=$LLVM_PREFIX/lib/cmake/mlir" >> $GITHUB_ENV
+  #
+  #      # set up ccache for faster C++ builds
+  #      - name: Setup ccache
+  #        uses: Chocobo1/setup-ccache-action@v1
+  #        with:
+  #          prepend_symlinks_to_path: false
+  #          override_cache_key: mlir-plugin-${{ matrix.runs-on }}-llvm-${{ matrix.llvm-version }}
+  #
+  #      # Set up uv for faster Python package management
+  #      - name: Install the latest version of uv
+  #        uses: astral-sh/setup-uv@v6
+  #        with:
+  #          python-version: 3.13
+  #          activate-environment: true
+  #
+  #      # Build the plugin package
+  #      - name: Build mqt-core-catalyst-plugin
+  #        working-directory: plugins/catalyst
+  #        run: uv sync --verbose --active
+  #
+  #        # Print shared library dependencies for the plugin
+  #      - name: Print shared library dependencies for mqt-core-catalyst-plugin
+  #        run: |
+  #          if [ "$(uname)" = "Linux" ]; then
+  #              ldd .venv/lib/python3.13/site-packages/mqt/core/plugins/catalyst/mqt-core-catalyst-plugin.so || true
+  #          elif [ "$(uname)" = "Darwin" ]; then
+  #              otool -l .venv/lib/python3.13/site-packages/mqt/core/plugins/catalyst/mqt-core-catalyst-plugin.dylib || true
+  #          fi
+  #
+  #      # Print shared library dependencies for Catalyst
+  #      - name: Print shared library dependencies for Catalyst
+  #        run: |
+  #          if [ "$(uname)" = "Linux" ]; then
+  #              ldd $(which catalyst) || true
+  #          elif [ "$(uname)" = "Darwin" ]; then
+  #              otool -l $(which catalyst) || true
+  #          fi
+  #
+  #      # Run the plugin tests
+  #      - name: Run pytest for mqt-core-catalyst-plugin
+  #        working-directory: plugins/catalyst
+  #        run: python -m pytest -s --verbose
```

Comment:
I'd still like to investigate at some point whether there is anything we can do to make this work with pre-compiled LLVM versions. But this will have to wait for a future PR. It would be great to create a tracking issue for this though.